packages/core/src/utils/security/index.ts

```diff
@@ -29,10 +29,10 @@ function getAllInvalidSec<R>(invalidSecuritySchemes: Array<Left<IPrismDiagnostic
   }
 }
 
-export function validateSecurity<R, I>(someInput: I, resource?: R): IPrismDiagnostic[] {
+export function validateSecurity<R, I>(someInput: I, resource?: R): IPrismDiagnostic | undefined {
   const securitySchemes = get(resource, 'security', []);
 
-  if (!securitySchemes.length) return [];
+  if (!securitySchemes.length) return;
```

Comment:
Why make `validateSecurity` return `IPrismDiagnostic | undefined` if it could be simpler to just have `IPrismDiagnostic[]`? What's wrong with an empty array? If we do `return this type or this type`, then we stray away from functional principles. If we really want to have `undefined` here, at least let's put it inside `Either`, so it's either `Right<IPrismDiagnostic>` or `Left<undefined>`.

Reply:
An array should represent the repetition of something; this is clearly not happening here. The ideal would be to return an `Option<IPrismDiagnostic>` — which is something I really want to start doing here in the codebase.

I will try to do this now but it might be painful and this needs to be rewritten anyway, the validation system is starting to kill us.

Reply:
@XVincentX, if `An array should represent the repetition of something`, then why it is allowed in some languages for an array to have misc types? Is `[1, 'abc', {}]` a repetition of what? A box of oranges (even if empty) is still a box of oranges. And a function's domain should never be a box of oranges and apples, just one of these

Reply:
@lag-of-death I've refactored the function to make it return an Option 796d18508bfa4d1279bd52498dd827ead2eeb4f7 — this should resolve this issue